Sources/FirebaseFirestore/DocumentReference+Swift.swift

```diff
@@ -64,6 +68,56 @@ extension DocumentReference {
 
     return ListenerRegistration(boxed, instance)
   }
+
+  public func setData(_ data: [String: Any], merge: Bool = false, completion: ((NSError?) -> Void)?) {
+    let boxed = Unmanaged.passRetained(completion as AnyObject)
+    let converted = FirestoreDataConverter.firestoreValue(document: data)
+    let options = merge ? firebase.firestore.SetOptions.Merge() : firebase.firestore.SetOptions()
+
+    Self.mutationQueue.async {
+      let future = swift_firebase.swift_cxx_shims.firebase.firestore.document_set_data(self, converted, options)
+
+      future.OnCompletion_SwiftWorkaround({ future, pvCallback in
+        guard let pCallback = pvCallback, let callback = Unmanaged<AnyObject>.fromOpaque(pCallback).takeUnretainedValue() as? ((NSError?) -> Void)? else {
+          return
+        }
+        if let code = future?.pointee.error(), code != 0 {
+          callback?(NSError(domain: "firebase.firestore.document", code: Int(code)))
+        } else {
+          callback?(nil)
+        }
+      }, UnsafeMutableRawPointer(boxed.toOpaque()))
+
+      future.Wait(firebase.FutureBase.kWaitTimeoutInfinite)
+    }
+  }
+}
+
+extension DocumentReference {
+  public func setData(_ data: [String: Any], merge: Bool = false) async throws -> Void {
```

Comment:
Void returns are implicit.

```suggestion
  public func setData(_ data: [String: Any], merge: Bool = false) async throws {
```